src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -95,6 +94,18 @@
  * address of the region of memory which backs the segment.</li>
  * </ul>
  * <p>
+ * Every memory segment has a {@linkplain #maxByteAlignment() maximum byte alignment}
+ * (see <a href="#segment-alignment">Alignment</a>"), expressed as a {@code long} value
+ * that is always an even power of two.
```

Comment:
I think it's better not to say too much about alignment here, otherwise we end up referring to concepts that are not well adequately introduced. Perhaps it would be better to just say:
```
Every memory segment has a maximum byte alignment, expressed as a long value. The maximum alignment is always a power of two, derived from the segment address, and the segment type, as explained in more details below <link>.
```